lib/rules/no-control-regex.js

```diff
@@ -83,8 +96,10 @@ module.exports = {
                 if (regex) {
                     const computedValue = regex.toString();
 
-                    if (hasControlCharacters(computedValue)) {
-                        context.report(node, "Unexpected control character in regular expression.");
+                    const controlCharacters = getControlCharacters(computedValue);
+
+                    if (controlCharacters.length > 0) {
+                        context.report(node, "Unexpected control character(s) in regular expression: " + controlCharacters.join(", ") + ".");
```

Comment:
@ilyavolodin Yes. I left a note in the tests, there are two test cases which report one lint message but have two characters in the message. We can only report one line/column (or a line/column start and line/column end, but still one range). Which column would be correct then, bearing in mind that the characters could be non-consecutive?

I would say the only way to safely report positions of violating characters is to report once per bad character. And as I said earlier, I don't think it's necessary just yet-- we can wait for end-user demand.